src/api-docs/source/index.html.md

```diff
@@ -45,18 +45,127 @@ search: true
 
 Welcome to the hsd API!
 
-The default hsd HTTP server listens on port (`12037` for main, `13037` for testnet, `14037` for regtest, and `15037` for simnet). It exposes a REST JSON, as well as a JSON-RPC API.
+hsd is a fork of [bcoin](https://github.com/bcoin-org/bcoin) and so if you are familiar with it,
+you'll find that the API is very similar to the [bcoin API](http://bcoin.io/api-docs/).
+bcoin was released with a JSON-RPC API that was based on the Bitcoin Core RPC interface.
+The actual Bitcoin Core RPC API has changed a lot since then, but if you are familiar with
+using bitcoind, you'll find a lot of the same RPC calls available in hsd.
+
+**There are four API servers in hsd**
+
+The full node and the wallet are separate processes in hsd, and their API servers run on separate ports.
+
+Network   | Wallet API Port | Node API Port
+--------- | --------------- | -------------
+main      | 12039           | 12037
+testnet   | 13039           | 13037
+regtest   | 14039           | 14037
+simnet    | 15039           | 15037
+
+In addition, each server has two API formats: JSON-RPC and a RESTful HTTP API.
+In general, the HTTP APIs respond to `GET` requests and the RPC APIs respond to
+`POST` requests. There are exceptions, so be sure to read the docs carefully.
+
+**Examples:**
+
+**Get a [transaction by hash](#get-tx-by-txhash) from the full node HTTP API:**
+
+<code>
+curl http://x:api-key@127.0.0.1:14037/tx/4674eb87021d9e07ff68cfaaaddfb010d799246b8f89941c58b8673386ce294f
+</code>
+
+**Get a [raw transaction by hash](#getrawtransaction) from the full node RPC API:**
+
+<code>
+curl http://x:api-key@127.0.0.1:14037
+  -X POST
+  --data '{
+    "method": "getrawtransaction",
+    "params": [ "4674eb87021d9e07ff68cfaaaddfb010d799246b8f89941c58b8673386ce294f"]
+  }'
+</code>
+
+**Get a [transaction by hash](#get-wallet-tx-details) from the "primary" wallet HTTP API:**
+
+<code>
+curl http://x:api-key@127.0.0.1:14039/wallet/primary/tx/4674eb87021d9e07ff68cfaaaddfb010d799246b8f89941c58b8673386ce294f
+</code>
+
+**Get a [transaction by hash](#gettransaction) from the wallet RPC API:**
+
+<code>
+curl http://x:api-key@127.0.0.1:14039
+  -X POST
+  --data '{
+    "method": "gettransaction",
+    "params": [ "4674eb87021d9e07ff68cfaaaddfb010d799246b8f89941c58b8673386ce294f" ]
+  }'
+</code>
+
+**Indexing**
+
+hsd has two indexer modules that are inactive by default. When turned on, the indexers
+enable additional API calls.
+
+Transaction indexer: enabled by `--index-tx=true`. Allows lookup of arbitrary transactions
+by their hash (txid). When disabled, the wallet still tracks all of its own transactions
+by hash and the full node can still be used to look up UTXO (unspent transaction outputs).
```

Comment:
The UTXOs are indexed by outpoint, which is the transaction ID that created the UTXO along with its index in that transaction.